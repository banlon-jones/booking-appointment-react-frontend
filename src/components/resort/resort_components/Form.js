/* eslint-disable react/prop-types */
import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import AddResort from '../AddResort.module.css';
import {
  selectResort,
  getResort,
} from '../../../store/resort-details/resortDetail';

// Button description: variable
let buttonDescription = '';

const Form = (props) => {
  const { formId } = props;
  const { addResort } = props;
  const { updateResortValue } = props;
  const { updateId } = useParams();

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getResort(updateId));
  }, []);

  const resortDetails = useSelector(selectResort);

  const setRestoreValue = (el) => {
    let result = '';
    if (formId !== 'add') {
      result = resortDetails[el];
    }
    return result;
  };

  if (formId === 'add') {
    buttonDescription = 'Add Resort';
  } else buttonDescription = 'Update Resort';

  return (
    <>
      <form
        className={AddResort.form}
        onSubmit={(e) => {
          if (formId === 'add') {
            addResort(e);
          } else updateResortValue(e);
        }}
        data-testid="form"
      >
        <input
          type="text"
          placeholder="Name"
          name="name"
          defaultValue={setRestoreValue('name')}
        />
        <input
          type="text"
          placeholder="Country"
          name="country"
          defaultValue={setRestoreValue('country')}
        />
        <input
          type="text"
          placeholder="City"
          name="city"
          defaultValue={setRestoreValue('city')}
        />
        <textarea
          placeholder="Resort description"
          col="50"
          row="50"
          name="description"
          defaultValue={setRestoreValue('description')}
        />

        <input
          type="text"
          placeholder="Cost"
          name="cost"
          defaultValue={setRestoreValue('cost')}
        />
        <div className={AddResort.file}>
          <span className={AddResort.label}>Choose a resort image:</span>
          <input
            className={AddResort.image_input}
            type="url"
            name="image"
            defaultValue={setRestoreValue('image')}
          />
        </div>
        <button type="submit" className={AddResort.button_resort_add}>
          {buttonDescription}
        </button>
      </form>
    </>
  );
};

export default Form;
