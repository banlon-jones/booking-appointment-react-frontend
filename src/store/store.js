/* eslint-disable import/no-named-as-default */
/* eslint-disable import/no-named-as-default-member */
import { configureStore } from '@reduxjs/toolkit';
import userSlice from './user/user';
import resortDetailSlice from './resort-details/resortDetail';

const store = configureStore({
  reducer: {
    user: userSlice,
    resortDetail: resortDetailSlice,
  },
});

export default store;
