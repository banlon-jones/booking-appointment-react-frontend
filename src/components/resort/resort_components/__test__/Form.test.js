/* eslint-disable comma-dangle */
import React from 'react';
import { cleanup, screen, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../../../store/store';
import Form from '../Form';

afterEach(cleanup);

describe('React Test for Form', () => {
  test('Form Test', () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <Form />
        </Provider>
      </BrowserRouter>
    );
    const ResortElement = screen.getByTestId('form');
    expect(ResortElement).toBeInTheDocument('Add Resort');
  });
});

describe('AddResort Snapshot', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <Form />
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
