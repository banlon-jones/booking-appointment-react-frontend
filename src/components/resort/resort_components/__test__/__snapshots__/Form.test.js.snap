// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddResort Snapshot renders correctly 1`] = `
<form
  className="form"
  data-testid="form"
  onSubmit={[Function]}
>
  <input
    name="name"
    placeholder="Name"
    required={true}
    type="text"
  />
  <input
    name="country"
    placeholder="Country"
    required={true}
    type="text"
  />
  <input
    name="city"
    placeholder="City"
    required={true}
    type="text"
  />
  <textarea
    col="50"
    name="description"
    placeholder="Resort description"
    required={true}
    row="50"
  />
  <input
    name="cost"
    placeholder="Cost"
    required={true}
    type="text"
  />
  <div
    className="file"
  >
    <span
      className="label"
    >
      Enter Resort Image URl:
    </span>
    <input
      className="image_input"
      name="image"
      required={true}
      type="url"
    />
  </div>
  <button
    className="button_resort_add"
    type="submit"
  >
    Update Resort
  </button>
</form>
`;
